osgp/protocol-adapter-mqtt/osgp-protocol-adapter-mqtt/src/main/java/org/opensmartgridplatform/adapter/protocol/mqtt/application/services/SubscriptionService.java

```diff
@@ -152,4 +146,17 @@ public void onReceive(final MqttClientAdapter mqttClientAdapter, final byte[] pa
             .build();
     this.outboundOsgpCoreResponseMessageSender.send(responseMessage);
   }
+
+  private int getDefaultPort() {
+    return Integer.valueOf(
+        this.mqttClientProperties.getProperty(MqttConstants.DEFAULT_PORT_PROPERTY_NAME));
```

Comment:
`Integer.parseInt` would fit the `int` result a little better.
Looks like a `NullPointerException` could occur is the `mqttClientProperties` did not have the specific property obtained here. These kind of things may be caught earlier with a more specific input object than `Properties` that could already do some validation on construction.